### PR TITLE
rllib changes to enable storing user_data in params.json

### PR DESCRIPTION
**Pull request by nskh**
## What do these changes do?

Modified rllib to allow `params.json` to store a dict of `user_data` by making tweaks to the PPO and DQN algorithms. Also fixed two typos. Changes to `.gitignore` and `result.py` can be excluded. 

  Sister PR: cathywu/learning-traffic#233

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,9 @@
 /src/local_scheduler/format/*_generated.h
 /src/ray/gcs/format/*_generated.h
 
+# Ray results should be ignored
```
Comment (cathywu) [inline .gitignore]: Please remove. `ray_results` does not default to being written inside the `ray` directory (as far as I know?).

File: python/ray/rllib/dqn/dqn.py [hunk 2]
```diff
@@ -102,7 +102,8 @@
     async_updates=False,
     # (Experimental) Whether to use multiple GPUs for SGD optimization.
     # Note that this only helps performance if the SGD batch size is large.
-    multi_gpu=False)
+    multi_gpu=False,
+    user_data={})
```
Comment (cathywu) [inline python/ray/rllib/dqn/dqn.py]: Please add a comment.

File: python/ray/rllib/eval.py [hunk 3]
```diff
@@ -29,14 +29,14 @@
 required_named.add_argument(
         "--run", type=str, required=True,
         help="The algorithm or model to train. This may refer to the name "
-        "of a built-on algorithm (e.g. RLLib's DQN or PPO), or a "
+        "of a built-in algorithm (e.g. RLLib's DQN or PPO), or a "
```
Comment (cathywu) [inline python/ray/rllib/eval.py]: Nice :)

File: python/ray/rllib/ppo/ppo.py [hunk 4]
```diff
@@ -81,7 +81,8 @@
     "write_logs": True,
     # Preprocessing for environment
     # TODO(rliaw): Convert to function similar to A#c
-    "preprocessing": {}
+    "preprocessing": {},
+    "user_data": {}
```
Comment (cathywu) [inline python/ray/rllib/ppo/ppo.py]: Again, please add a comment

File: python/ray/tune/result.py [hunk 5]
```diff
@@ -22,7 +22,7 @@
 """
 
 # Where ray.tune writes result files by default
-DEFAULT_RESULTS_DIR = os.path.expanduser("~/ray_results")
+DEFAULT_RESULTS_DIR = os.path.expanduser("~/Development/research/ray/ray_results")
```
Comment (cathywu) [inline python/ray/tune/result.py]: Please don't include this change.

**Review by cathywu — CHANGES_REQUESTED** [review]

**Comment by cathywu** [comment]
Yay, I like this clean solution. I just left some cleanup comments.

**Comment by nskh** [comment]
Updated in accordance with your requested changes; should be good to go.

**Review by cathywu — APPROVED** [review]

**Pull request merged**